blob/driver/driver.go

```diff
@@ -304,3 +306,56 @@ type SignedURLOptions struct {
 	// Expiry sets how long the returned URL is valid for. It is guaranteed to be > 0.
 	Expiry time.Duration
 }
+
+// prefixedBucket implements Bucket by prepending prefix to all keys.
+type prefixedBucket struct {
+	base   Bucket
+	prefix string
+}
+
+// NewPrefixedBucket returns a Bucket based on b with all keys modified to have
+// prefix.
+func NewPrefixedBucket(b Bucket, prefix string) Bucket {
```

Comment:
I don't understand why we're exporting this function in the driver. Can you help me understand the use case?

Reply:
It is exported from the driver so that `blob.go` can call it.

We could alternatively move the `prefixedBucket` type into `blob.go` entirely, but
a) I felt like it was more natural here.
b) The conformance tests are run given a `driver.Bucket`, so we need to be able to wrap a `driver.Bucket` in order to run the conformance test suite on a prefixed bucket.

An alternative might be to add a conformance test that starts with a `driver.Bucket`, makes a `blob.Bucket` out of it, and then turns it into a prefixed `Bucket`, and then runs a test. This would make it so that every driver was tested with a prefixed bucket instead of just `mem` + `file`, but would mean that we only run a specific conformance test (not yet written), as opposed to what happens now, which is that the entire conformance test suite is run against a prefixed bucket.